src/dbnode/network/server/tchannelthrift/node/service.go

```diff
@@ -514,10 +516,16 @@ func (s *service) query(ctx context.Context, db storage.Database, req *rpc.Query
 	if req.NoData != nil && *req.NoData {
 		fetchData = false
 	}
+	reader := docs.NewEncodedDocumentReader()
```

Comment:
Calling this method out as something to review closely. Want to ensure I'm thinking about this correctly.